services/src/test/java/org/apache/druid/server/AsyncQueryForwardingServletTest.java

```diff
@@ -1063,4 +942,41 @@ public int getPort()
       return port;
     }
   }
+  private ArgumentCaptor<Exception> getArgumentCaptor(ServerConfig serverConfig, String errorMessage, boolean haverequest) throws IOException
+    {
+    ObjectMapper mockMapper = Mockito.mock(ObjectMapper.class);
+    HttpServletResponse response = Mockito.mock(HttpServletResponse.class);
+    ServletOutputStream outputStream = Mockito.mock(ServletOutputStream.class);
+    Mockito.when(response.getOutputStream()).thenReturn(outputStream);
+    final AsyncQueryForwardingServlet servlet = new AsyncQueryForwardingServlet(
+            new MapQueryToolChestWarehouse(ImmutableMap.of()),
+            mockMapper,
+            TestHelper.makeSmileMapper(),
+            null,
+            null,
+            null,
+            new NoopServiceEmitter(),
+            new NoopRequestLogger(),
+            new DefaultGenericQueryMetricsFactory(),
+            new AuthenticatorMapper(ImmutableMap.of()),
+            new Properties(),
+            serverConfig
+    );
+    Exception testException = new IllegalStateException(errorMessage);
+
+    if (haverequest) {
+      HttpServletRequest request = Mockito.mock(HttpServletRequest.class);
+      Mockito.when(request.getAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT)).thenReturn(new AuthenticationResult("userA", "basic", "basic", null));
+      IOException testException2 = new IOException(errorMessage);
+      servlet.handleQueryParseException(request, response, mockMapper, testException2, false);
```

Comment:
I would prefer using the originally suggested lambda style.
While reading the main test, it should be evident which method of `AsyncQueryForwardingServlet` is being called.

Also, the exception being handled should be _passed_ to this method rather than being constructed here.